packages/menu/src/test/mwc-menu.test.ts

```diff
@@ -0,0 +1,542 @@
+      const surface =
+          element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+      assert.equal(element.fixed, false);
+      assert.equal(surface.fixed, false);
+      element.fixed = true;
+      await element.updateComplete;
+      assert.equal(element.fixed, true);
+      assert.equal(surface.fixed, true);
+    });
+
+    test('`fullwidth` is passed to surface', async () => {
+      const surface =
+          element.shadowRoot!.querySelector<MenuSurface>('.mdc-menu')!;
+      assert.equal(element.fullwidth, false);
+      assert.equal(surface.fullwidth, false);
+      element.fullwidth = true;
+      await element.updateComplete;
+      assert.equal(element.fullwidth, true);
+      assert.equal(surface.fullwidth, true);
+    });
+
+    test('correct roles are passed to inner list', async () => {
+      const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+      assert.equal(element.innerRole, 'menu');
+      assert.equal(list.innerRole, 'menu');
+      assert.equal(list.itemRoles, 'menuitem');
+      element.innerRole = 'listbox';
+      await element.updateComplete;
+      assert.equal(list.innerRole, 'listbox');
+      assert.equal(list.itemRoles, 'option');
+    });
+
+    test('`activatable` is set on inner list', async () => {
+      const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+      assert.equal(element.activatable, false);
+      assert.equal(list.activatable, false);
+      element.activatable = true;
+      await element.updateComplete;
+      assert.equal(element.activatable, true);
+      assert.equal(list.activatable, true);
+    });
+  });
+
+  suite('multi', () => {
+    setup(async () => {
+      fixt = await fixture(menu({
+        multi: true,
+        open: true,
+        contents: html`
+          <mwc-list-item>1</mwc-list-item>
+          <mwc-list-item>2</mwc-list-item>`
+      }));
+      element = fixt.root.querySelector('mwc-menu')!;
+      await element.updateComplete;
+    });
+
+    test('`multi` is set on inner list', async () => {
+      const list = element.shadowRoot!.querySelector<List>('.mdc-list')!;
+      assert.equal(element.multi, true);
+      assert.equal(list.multi, true);
+      element.multi = false;
+      await element.updateComplete;
+      assert.equal(element.multi, false);
+      assert.equal(list.multi, false);
+    });
+
+    test('clicking items sets selection', async () => {
+      const item0 = element.children[0] as ListItem;
+      const item1 = element.children[1] as ListItem;
+      item0.click();
+      item1.click();
+      const items = element.selected!;
+      assert.equal(items[0], item0);
+      assert.equal(items[1], item1);
+    });
+  });
+
+  suite('grouped', () => {
+    let oldSetTimeout;
+
+    setup(async () => {
+      oldSetTimeout = window.setTimeout;
+      // TODO (43081j): mock this properly
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
@e111077 here shall i just use sinon for now and hope my pr to make their fake-timers lib work comes soon? so then we can drop sinon but keep the mock timers

Reply:
If you use sinon now, wouldn't it just not work until your PR comes in? I think this TODO is okay for now.

Reply:
What they've done is published the timers part of it as a standalone package. But it turns out what they published is broken for esm consumers and bundlers. So I couldn't use it here. 

However, we still have Sinon as a whole until that gets fixed (then we can drop Sinon and keep just the timers). Sinon wraps the timers module, so we could use it here until the standalone module works..

`sinon.useFakeTimers()`. I've done it in another file somewhere around here in this pr.

Reply:
ah that works. go for it